tools/zim_install

```diff
@@ -0,0 +1,158 @@
+  if [ -e "${HOME}/.zlogin" ]; then
+    printf "${yellow}Found ${HOME}/.zlogin!$ ${blue}Backing up to ${HOME}/.zlogin_zimbackup${normal}\n"
+    mv "${HOME}/.zlogin" "${HOME}/.zlogin_zimbackup"
+  fi
+
+  # Place template files in ~/
+  cp -a "${HOME}/.zim/templates/" "${HOME}/"
+
+  # If this user's login shell is not already "zsh", attempt to switch
+  if [ ! "$(expr "${SHELL}" : '.*/\(.*\)')" = "zsh" ]; then
+    # Auto-change shell if platform provides "chsh" command
+    if hash chsh >/dev/null 2>&1; then
+      printf "${blue}Changing your default shell to zsh!${normal}\n"
+      chsh -s $(grep "/zsh$" "/etc/shells" | tail -1)
+    # If not, suggest user do so manually.
+    else
+      printf "${yellow}I can't change your shell automatically because this system does not have chsh.${normal}\n"
+      printf "${yellow}Please manually change your default shell to zsh!${normal}\n"
+    fi
+  fi
+
+  Z="${red}Z${green}"
+  printf "${green}   ______    _       _____ __  __                              _               ${normal}\n"
+  printf "${green}  |$Z$Z$Z$Z$Z/   | |     |$Z$Z$Z$Z$Z|$Z$Z\/$Z$Z|                            | |${normal}\n"
+  printf "${green}     /$Z/ ___| |__     |$Z| |$Z\\$Z$Z/$Z|_ __  _ __ _____   _____  __| |       ${normal}\n"
+  printf "${green}    /$Z/ / __| '_ \    |$Z| |$Z|\/|$Z| '_ \| '__/ _ \ \ / / _ \/ _\` |          ${normal}\n"
+  printf "${green}   /$Z/__\__ | | | |  _|$Z|_|$Z|  |$Z| |_) | | | (_) \ V |  __| (_| |          ${normal}\n"
+  printf "${green}  /$Z$Z$Z$Z$Z|___|_| |_| |$Z$Z$Z$Z$Z|$Z|  |$Z| .__/|_|  \___/ \_/ \___|\__,_|  ${normal}\n"
+  printf "${green}                                 | |                                           ${normal}\n"
+  printf "${green}                                 |_|                                           ${normal}\n"
+  unset Z
+
+  # For archival purposes
+  #   ______      _       _____  __  __                                    _
+  #  |___  /     | |     |_   _||  \/  |                                  | |
+  #     / /  ___ | |__     | |  | \  / | _ __   _ __  ___ __   __ ___   __| |
+  #    / /  / __|| '_ \    | |  | |\/| || '_ \ | '__|/ _ \\ \ / // _ \ / _` |
+  #   / /__ \__ \| | | |  _| |_ | |  | || |_) || |  | (_) |\ V /|  __/| (_| |
+  #  /_____||___/|_| |_| |_____||_|  |_|| .__/ |_|   \___/  \_/  \___| \__,_|
+  #                                      | |
+  #                                      |_|
+
+  # Remind user to copy back old settings
+  if [ -f "${HOME}/.zshrc_zimbackup" ] || [ -L "${HOME}/.zshrc_zimbackup" ]; then
+    printf "${yellow}Please inspect ${HOME}/.zshrc_zimbackup and copy over${normal}\n"
+    printf "${yellow}wanted settings to ${HOME}/.zshrc${normal}\n"
+  fi
+  if [ -f "${HOME}/.zlogin_zimbackup" ] || [ -L "${HOME}/.zlogin_zimbackup" ]; then
+    printf "${yellow}Please inspect ${HOME}/.zlogin_zimbackup and copy over${normal}\n"
+    printf "${yellow}wanted settings to ${HOME}/.zlogin${normal}\n"
+  fi
+
+  # Load zsh
+  env zsh
```

Comment:
That would lead me to believe that just calling the `exec` builtin should be fine, but of course, that will need to be tested. I don't see why it wouldn't work, though.